addons/test_mail/tests/test_mail_channel.py

```diff
@@ -226,6 +226,32 @@ def test_channel_auto_unsubscribe_archived_or_deleted_users(self):
         self.assertEqual(test_channel_group.channel_partner_ids, self.env['res.partner'])
         self.assertEqual(self.test_channel.channel_partner_ids, self.user_employee.partner_id | test_partner)
 
+    def test_channel_get(self):
+        current_user = self.env['res.users'].create({
+            "login": "adam",
+            "name": "Jonas",
+        })
+        current_user = current_user.with_user(current_user)
+        current_partner = current_user.partner_id
+        other_partner = self.test_partner
+
+        # `channel_get` should return a new channel the first time a partner is given
+        initial_channel_info = current_user.env['mail.channel'].channel_get(partners_to=other_partner.ids)
+        self.assertEqual(set(p['id'] for p in initial_channel_info['members']), {current_partner.id, other_partner.id})
+
+        # `channel_get` should return the existing channel every time the same partner is given
+        same_channel_info = current_user.env['mail.channel'].channel_get(partners_to=other_partner.ids)
+        self.assertEqual(same_channel_info['id'], initial_channel_info['id'])
+
+        # `channel_get` should return the existing channel when the current partner is given together with the other partner
+        together_channel_info = current_user.env['mail.channel'].channel_get(partners_to=(current_partner + other_partner).ids)
+        self.assertEqual(together_channel_info['id'], initial_channel_info['id'])
+
+        # `channel_get` should return a new channel when just the current partner is given
+        solo_channel_info = current_user.env['mail.channel'].channel_get(partners_to=current_partner.ids)
+        self.assertNotEqual(solo_channel_info['id'], initial_channel_info['id'])
+        self.assertEqual(set(p['id'] for p in solo_channel_info['members']), {current_partner.id})
```

Comment:
I don't think that's what we want, to always create a new channel with `channel_get` when just the current partner is provided. Seems more natural that `channel_get` with just current partner is equivalent to retrieving chat with self, so I would expect it to return existing chat if it has already been created.

Reply:
Comment is not clear but what you say is correct. This test line is the first time so it returns a new one, test should have another test to ensure it returns the same chat when calling again.

Reply:
TBH not super confident with SQL, but at a glance it looks like it will create another chat when you call `channel_get` with just current partner again?

I've put comment here because this part of test let me assume it was intended to always create `channel_get` in this condition, which felt wrong for me.

Reply:
It worked as expected already. I clarified the test comment and added the other test case.